python/rmm/_lib/device_buffer.pyx

```diff
@@ -57,17 +59,36 @@ cdef class DeviceBuffer:
         buf.c_obj = move(ptr)
         return buf
 
+    @staticmethod
+    @cython.boundscheck(False)
+    cdef DeviceBuffer c_frombytes(const unsigned char[::1] b,
+                                  uintptr_t stream=0):
```

Comment:
Do we need to mark this as `except *` or because we're returning a `DeviceBuffer` Cython can handle the possible Python exception?

Reply:
AIUI Cython extension types (like `DeviceBuffer`) are Python objects. As such any exception would be propagated in the return value (and Cython does generate code to check for this automatically).